android/src/main/java/com/magicleap/magicscript/scene/nodes/layouts/PageViewNode.kt

```diff
@@ -49,32 +47,41 @@ class PageViewNode(props: ReadableMap, layoutManager: LayoutManager<PageViewLayo
     init {
         // set default values of properties
         properties.putDefault(PROP_ALIGNMENT, DEFAULT_ALIGNMENT)
-        properties.putDefault(PROP_CONTENT_ALIGNMENT, DEFAULT_CONTENT_ALIGNMENT)
-        properties.putDefault(PROP_PADDING, DEFAULT_ITEM_PADDING)
+        properties.putDefault(PROP_DEFAULT_CONTENT_ALIGNMENT, DEFAULT_CONTENT_ALIGNMENT)
+        properties.putDefault(PROP_DEFAUL_PAGE_PADDING, DEFAULT_ITEM_PADDING)
         properties.putDefault(PROP_VISIBLE_PAGE, DEFAULT_VISIBLE_PAGE)
     }
 
     override fun applyProperties(props: Bundle) {
         super.applyProperties(props)
 
-        if (properties.containsAny(PROP_VISIBLE_PAGE, PROP_PADDING, PROP_CONTENT_ALIGNMENT)) {
+        if (properties.containsAny(
+                PROP_VISIBLE_PAGE,
+                PROP_DEFAUL_PAGE_PADDING,
+                PROP_DEFAULT_CONTENT_ALIGNMENT
+            )
+        ) {
             requestLayout()
         }
     }
 
     override fun getLayoutParams(): PageViewLayoutParams {
         val visiblePage = properties.getDouble(PROP_VISIBLE_PAGE, DEFAULT_VISIBLE_PAGE).toInt()
-        val padding = properties.read<Padding>(PROP_PADDING)!!
-        val contentAlignment = properties.read<Alignment>(PROP_CONTENT_ALIGNMENT)!!
-        val contentHorizontalAlignment = contentAlignment.horizontal
-        val contentVerticalAlignment = contentAlignment.vertical
+        val defaultItemsPadding = properties.read<Padding>(PROP_DEFAUL_PAGE_PADDING)!!
+        val defaultItemsAlignment = properties.read<Alignment>(PROP_DEFAULT_CONTENT_ALIGNMENT)!!
+        val itemsPadding = properties.read<ItemPaddingMap>(UiLinearLayout.PROP_ITEM_PADDING)
+        val itemsAlignment = properties.read<ItemAlignmentMap>(UiLinearLayout.PROP_ITEM_ALIGNMENT)
+
+        val childrenPadding =
+            calculateChildrenPadding(defaultItemsPadding, itemsPadding?.paddings)
+        val childrenAlignment =
+            calculateChildrenAlignment(defaultItemsAlignment, itemsAlignment?.alignments)
```

Comment:
Using the `buildMap` method it would be like: 
```
val keys= (0 until childrenList.size).toList()
val childrenPadding = itemsPadding?: buildMap(keys, defaultItemsPadding)
```